engine/src/main/java/org/entando/entando/aps/system/services/dataobjectmodel/DataObjectModelManager.java

```diff
@@ -39,178 +39,185 @@
+			_logger.error("Error deleting a dataObject model", t);
+			throw new ApsSystemException("Error deleting a dataObject model", t);
+		}
+	}
+
+	@Override
+	public void updateDataObjectModel(DataObjectModel model) throws ApsSystemException {
+		try {
+			this.getDataModelDAO().updateDataModel(model);
+			this._dataModels.put(new Long(model.getId()), model);
+			this.notifyDataModelChanging(model, DataObjectModelChangedEvent.UPDATE_OPERATION_CODE);
+		} catch (Throwable t) {
+			_logger.error("Error updating a dataObject model", t);
+			throw new ApsSystemException("Error updating a dataObject model", t);
+		}
+	}
+
+	private void notifyDataModelChanging(DataObjectModel dataObjectModel, int operationCode) throws ApsSystemException {
+		DataObjectModelChangedEvent event = new DataObjectModelChangedEvent();
+		event.setDataObjectModel(dataObjectModel);
+		event.setOperationCode(operationCode);
+		this.notifyEvent(event);
+	}
+
+	@Override
+	public DataObjectModel getDataObjectModel(long dataObjectModelId) {
+		return (DataObjectModel) _dataModels.get(new Long(dataObjectModelId));
+	}
+
+	@Override
+	public List<DataObjectModel> getDataObjectModels() {
+		List<DataObjectModel> models = new ArrayList<DataObjectModel>(this._dataModels.values());
+		Collections.sort(models);
+		return models;
+	}
+
+	@Override
+	public List<DataObjectModel> getModelsForDataObjectType(String dataType) {
+		List<DataObjectModel> models = new ArrayList<DataObjectModel>();
+		Object[] allModels = this._dataModels.values().toArray();
+		for (int i = 0; i < allModels.length; i++) {
+			DataObjectModel dataObjectModel = (DataObjectModel) allModels[i];
+			if (null == dataType || dataObjectModel.getDataType().equals(dataType)) {
+				models.add(dataObjectModel);
+			}
+		}
+		return models;
+	}
+
+	@Override
+	public Map<String, List<IPage>> getReferencingPages(long modelId) {
+		Map<String, List<IPage>> utilizers = new HashMap<String, List<IPage>>();
+		IPage root = this.getPageManager().getDraftRoot();
+		this.searchReferencingPages(modelId, root, utilizers);
+		root = this.getPageManager().getOnlineRoot();
+		this.searchReferencingPages(modelId, root, utilizers);
+		return utilizers;
+	}
+
+	private void searchReferencingPages(long modelId, IPage page, Map<String, List<IPage>> utilizers) {
+		this.addReferencingPage(modelId, page, utilizers);
+		boolean isOnline = page.isOnline();
+		String[] children = page.getChildrenCodes();
+		for (int i = 0; i < children.length; i++) {
+			IPage child = (isOnline)
+					? this.getPageManager().getOnlinePage(children[i])
+					: this.getPageManager().getDraftPage(children[i]);
+			if (null != child) {
+				this.searchReferencingPages(modelId, child, utilizers);
+			}
+		}
+	}
+
+	private void addReferencingPage(long modelId, IPage page, Map<String, List<IPage>> utilizers) {
+		if (null != page && null != page.getWidgets()) {
+			Widget[] widgets = page.getWidgets();
+			for (int i = 0; i < widgets.length; i++) {

```

Comment:
Are we officially on Java8? This hole loop/check/set structure would be a pretty good usecase for a stream/filter/lamda combo.

Reply:
At the moment we use java7 as default reference